gpjax/parameters.py

```diff
@@ -1,27 +1,64 @@
 import typing as tp
 import warnings
+from collections import namedtuple
 from copy import deepcopy
+from warnings import warn
 
 import distrax as dx
 import jax
 import jax.numpy as jnp
+import jax.random as jr
+from chex import dataclass
 from jaxtyping import f64
 from tensorflow_probability.substrates.jax import distributions as tfd
 
 from .config import get_defaults
+from .utils import merge_dictionaries
 
 Identity = dx.Lambda(lambda x: x)
 
 
 ################################
 # Base operations
 ################################
-def initialise(obj) -> tp.Tuple[tp.Dict, tp.Dict, tp.Dict]:
+@dataclass
+class ParameterState:
+    """The state of the model. This includes the parameter set and the functions that allow parameters to be constrained and unconstrained."""
+
+    params: tp.Dict
+    trainables: tp.Dict
+    constrainers: tp.Dict
+    unconstrainers: tp.Dict
+
+    def unpack(self):
+        return self.params, self.trainables, self.constrainers, self.unconstrainers
+
+
+def initialise(model, key: jnp.DeviceArray = None, **kwargs) -> ParameterState:
     """Initialise the stateful parameters of any GPJax object. This function also returns the trainability status of each parameter and set of bijectors that allow parameters to be constrained and unconstrained."""
-    params = obj.params
+    if key is None:
+        warn("No PRNGKey specified. Defaulting to seed 123.")
+        key = jr.PRNGKey(123)
+    params = model._initialise_params(key)
```

Comment:
Good. In future, however, we should not raise a warning if the model does not require a key.